client/src/pages/User/Profile.js

```diff
@@ -0,0 +1,171 @@
+    } else {
+      setSnack({ type: "success", message: "Success", open: true });
+    }
+  };
+
+  const handleSubmission = async () => {
+    setLoading(true);
+    const formData = new FormData();
+    formData.append("file", selectedFile);
+
+    try {
+      const uploadResponse = await (
+        await fetch("/api/file-upload", {
+          method: "POST",
+          body: formData,
+        })
+      ).json();
+      openSnack(!uploadResponse.imageUrl ? "Upload failed!" : null);
+      if (uploadResponse.imageUrl) {
+        setAvatarImage(uploadResponse.imageUrl);
+        setOpen(false);
+      }
+      setLoading(false);
+      setSelectedFile(null);
+    } catch (error) {
+      setLoading(false);
+      openSnack(error.message);
+    }
+  };
+
+  const handleDrop = (files) => {
+    files[0].preview = URL.createObjectURL(files[0]);
+    setSelectedFile(files[0]);
+  };
+
+  const closeSnack = () => {
+    setSnack((prevState) => {
+      return { ...prevState, open: false };
+    });
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <Grid className={classes.root}>
+      <FileUploaderDialog
+        file={selectedFile}
+        open={open}
+        loading={loading}
+        close={handleClose}
+        submit={handleSubmission}
+        handleDrop={handleDrop}
+      />
+      <Drawer
+        className={classes.drawer}
+        variant="permanent"
+        classes={{
+          paper: classes.drawerPaper,
+        }}
+      >
+        <Grid className={classes.profilePosition}>
+          <Avatar
+            component={ButtonBase}
+            src={avatarImage}
+            className={classes.avatar}
+            onClick={handleClickOpen}
+          />
+          <Typography variant="h6">{mockUser.name}</Typography>
+          <Typography className={classes.email}>{mockUser.email}</Typography>
+
+          <Grid className={classes.editBtnContainer}>
+            <Button className={classes.editBtn} variant="outlined">
+              Edit
+            </Button>
+          </Grid>
+        </Grid>
+        <Grid className={classes.drawerContainer}>
+          <ul>
+            <NavLink
+              activeStyle={{ color: "black", borderLeft: "3px solid #FFA000" }}
+              className={classes.profileLinks}
+              to="/profile/favoritedestinations"
+            >
+              <li>Favorite Destinations</li>
+            </NavLink>
+            <NavLink
+              activeStyle={{ color: "black", borderLeft: "3px solid #FFA000" }}
+              className={classes.profileLinks}
+              to="/profile/notifications"
+            >
+              <li>Notifications</li>
+            </NavLink>
+            <NavLink
+              activeStyle={{ color: "black", borderLeft: "3px solid #FFA000" }}
+              className={classes.profileLinks}
+              to="/profile/accountsettings"
+            >
+              <li>Account Settings</li>
+            </NavLink>
```

Comment:
Check out the docs on the [`NavLink`](https://reactrouter.com/web/api/NavLink/activeclassname-string) component, especially in relation to the `activeClassName` that is provided out of the box. It may help you reduce the duplication that you need to add if you're using the `activeStyle` prop.